test/test_nn.py

```diff
@@ -285,6 +298,20 @@ def _get_target(self):
 
 
 class TestNN(NNTestCase):
+    # Used for run the same test with different tensor types
+    def repeat_test_for_types(dtypes):
```

Comment:
If you want to put it as a regular function, but in a class, then you need to decorate it with `@staticfunction`